File: Tema 3/SpaceshipsShop/SpaceshipsShop.Lib/Models/SpaceShip.cs
using System;

namespace SpaceshipsShop.Lib.Models
{
    public class SpaceShip
    {
        public Guid Id { get; set; }

        public string Brand { get; set; }

        public string Password { get; set; }

        public int FTLFactor { get; set; }

        public string Color { get; set; }

        public int PassengersCapacity { get; set; }
    }
}

File: Tema 3/SpaceshipsShop/SpaceshipsShop.Web/wwwroot/scripts/Views/Spaceship/Spaceship.js
class Spaceship
{
    Http = null;

    GridOptions = 
    {
        columnDefs : 
        [ 
                { name: 'Brand', field: 'Brand'},
                { name: 'FTLFactor', field: 'FTLFactor'},
                { name: 'Color', field: 'Color'},
                { name: 'PassengersCapacity', field: 'PassengersCapacity'},
        ],
        data : []
      };

    
    constructor($http)
    {        
        this.Http = $http;
        this.GetAll();
    }

    GetAll()
    {
        this.Http(
        {
            method: 'GET',            
            url: 'api/spaceships'            
        })
        .then((response) => this.LoadShips(response.data), 
        function error(e) 
        {
            alert(e.statusText);
        });
    }

    LoadShips(ships)
    {
        for (let i in ships)
            this.GridOptions.data.push(ships[i]);
    }
}

App.
  component('spaceship', {   
    templateUrl: 'scripts/views/spaceship/spaceship.html',
    controller: Spaceship,
    controllerAs: "vm"
  });